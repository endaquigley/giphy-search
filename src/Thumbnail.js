import React from 'react';
import styled from 'styled-components';

const GridChild = styled.div`
  /* perserve aspect ratio in Firefox */
`;

const Square = styled.div`
  position: relative;
  cursor: pointer;
  overflow: hidden;
  transform: scale(1);
  will-change: transform;
  transition: transform 0.3s ease;
  background-color: rgba(0, 0, 0, 0.05);
  border: 1px solid rgba(0, 0, 0, 0.30);
  padding-bottom: 100%;
  &:hover {
    transform: scale(0.9);
  }
`;

const Image = styled.img`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  object-fit: cover;
`;

const Thumbnail = ({ source, handleClick }) => {
  return (
    <GridChild>
      <Square>
        <Image
          src={ source }
          onClick={ () => handleClick() }
        />
      </Square>
    </GridChild>
  )
};

export default Thumbnail;
