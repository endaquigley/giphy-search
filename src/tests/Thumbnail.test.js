import React from "react";
import { render } from "react-testing-library";

import Thumbnail from "../Thumbnail";

import "react-testing-library/cleanup-after-each";

describe("Thumbnail Component", () => {
  it("renders an image element with the correct src", () => {
    const source = "https://enda.ie/filename-01.gif";
    const { container } = render(<Thumbnail source={source} />);

    const image = container.querySelector("img");
    expect(image.src).toBe("https://enda.ie/filename-01.gif");

    expect(container).toMatchSnapshot();
  });
});
