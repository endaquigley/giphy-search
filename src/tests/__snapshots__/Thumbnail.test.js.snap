// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an image element with the correct src 1`] = `
<div>
  <div
    class="sc-bdVaJa kiUbly"
  >
    <div
      class="sc-bwzfXH fFwymd"
    >
      <img
        class="sc-htpNat jtiZZo"
        src="https://enda.ie/filename-01.gif"
      />
    </div>
  </div>
</div>
`;
